packages/vite/src/node/optimizer/scan.ts

```diff
@@ -48,8 +48,11 @@ export const importsRE =
   /(?<!\/\/.*)(?<=^|;|\*\/)\s*import(?!\s+type)(?:[\w*{}\n\r\t, ]+from)?\s*("[^"]+"|'[^']+')\s*(?=$|;|\/\/|\/\*)/gm
 
 export async function scanImports(config: ResolvedConfig): Promise<{
```

Comment:
Is the return type right here? It looks like its returning `{ cancel, result }` and not `Promise<{ cancel, result }>`

Reply:
I was following @dominikg's PR here. The `esbuildContext` is only available after waiting for other async operations. But I agree that from a consumer POV, it is better to avoid the double promise, [vitejs/vite@`5ff8d8b` (#12225)](https://github.com/vitejs/vite/pull/12225/commits/5ff8d8b5942860e1d897ebd825cd085acc6e58f6) avoids this at the expense of a bit more complexity inside `scanImports`